File: JAVA/ChanUkNoh/java_work/src/Thirteenth/IteratorTest.java
package Thirteenth;

import java.util.ArrayList;
import java.util.Iterator;

public class IteratorTest {
    public static void main(String[] args) {
        // ArrayList<>:
        // ArrayList 는 다양한 데이터 타입을 수용할 수 있게해준다.
        // <String> 문자열, <Integer> 정수, <Float> float 타입 등등...
        ArrayList<String> list1 = new ArrayList<String >();
        ArrayList<Integer> list2 = new ArrayList<Integer>();

        list1.add("one");
        list1.add("two");
        list1.add("three");
        list1.add("four");
        list1.add("five");

        String s;
        Iterator e = list1.iterator();

        while (e.hasNext()) {
            s = (String) e.next();
            System.out.println(s);
        }
    }
}

File: JAVASCRIPT/ParkJaemin/js_test/src/210203/iterator-test/IteratorTest.js
class IteratorManager{
    constructor() { //클래스 생성
        this.itrData = [] // 배열 초기화
    }
    add(data){
        this.itrData.push(data) //data push
    }
    [Symbol.iterator](){
        return this.itrData.values() // value값 리턴

    }
}

const IteratorTest = () => {
    let iterData = ["one", "two", "three"]

    for(let entry of iterData){
        console.log(entry)
    }

    let varData = ["four", "five", "six"]

    let itr = varData[Symbol.iterator]()

    console.log(itr.next())
    console.log(itr.next())
    console.log(itr.next()) // 값이 있으면 done은 false
    console.log(itr.next()) // 값이 없으니 value는 undefined // done은 true
    // value랑 done은 반대로 나타난다.

    let varItr = varData.values() // 심볼 미사용

    console.log(varItr.next())
    console.log(varItr.next())
    console.log(varItr.next())
    console.log(varItr.next())

    return (
        <div className="IteratorTest">
            <p>
                IteratorTest
            </p>
        </div>
    )
}

export default IteratorTest